### DISCUSSION TITLE
Cancel in-flight effects at IdentifiedArray or StackState

### DISCUSSION THREAD

**Discussion by devkimmy** [pointfreeco/swift-composable-architecture#3532] [Q&A]
First of all, I appreciate your great contributes for this awesome Framework.😁

We used to declare `IndentifiedArray`  for representing List types, or declare `StackState` for representing Navigation.
And there might be `in-flight effect` running inside each items(Both of IndentifiedArray's State, StackState's State)

We sometimes face the moments for **removal of items**, compare to 'PresentationState' 's dismiss mechanism (using DismissEffect) there seems no such mechanism which automatically cancels in-flight effects.

Such as bellow

```swift
SomeReducer {
  Reduce { state, action in 
  ...
  case .removeItem(let item):
    // Consider `items: IdentifiedArrayOf<SomeMyReducer.State>`
    state.items.remove(item) // <- Do not cancels it's own in-flight effects
    return .none
  }
}
```

Should we always cancel item's in-flight effects manually? 
(Trigger actions of child which cancels all in-flight effects through `.cancel` effect)
Or is there any mechanism that I don't know?

Thank you!

**Comment by mbrandonw** [comment]
Hi @devkimmy, providing a minimal sample project that demonstrates the problem is the easiest way for us to understand the problem you are having.

Are you using `forEach` in order to integrate the `SomeMyReducer` into the parent domain? If so, then removing elements from the array should cancel the effects. If you are seeing otherwise, a project would be helpful.
